### PR TITLE
Add IP and agent in trace for failed login

### LINKED ISSUE #16423: Log the client IP address and the agent when login fail
Sometimes some pipeline or malicious tool uses invalid password to log in Harbor server, there is no log information to find out the source of this failed login. need to log the client IP and the agent.

### PR DESCRIPTION
**Pull request by stonezdj**
Fixes #16423

Signed-off-by: stonezdj <stonezdj@gmail.com>

### CODE REVIEW

**Comment by chlins** [comment]
@stonezdj Please rebase the latest main to resolve the CodeQL alerts.

File: src/server/middleware/security/basic_auth.go [hunk 1]
```diff
@@ -26,6 +26,23 @@ import (
 
 type basicAuth struct{}
 
+// GetIP get ip from request
+func GetIP(r *http.Request) string {
```
Comment (yxxhero) [inline src/server/middleware/security/basic_auth.go:30]: 1. GetIP --> GetClientIP  is better?
2. require unittest for this func.

File: src/server/middleware/security/basic_auth.go [hunk 2]
```diff
@@ -26,6 +26,23 @@ import (
...
+	}
+	forwarded := r.Header.Get("x-forwarded-for")
+	if forwarded != "" {
+		return forwarded
+	}
+	return r.RemoteAddr
+}
+
+// GetUserAgent get the user agent of current request
+func GetUserAgent(r *http.Request) string {
```
Comment (yxxhero) [inline src/server/middleware/security/basic_auth.go:42]: require unittest for this func.

**Review by chlins — APPROVED** [review]
lgtm

**Review by wy65701436 — APPROVED** [review]
lgtm

**Pull request merged**